src/wallet/wallet.ts

```diff
@@ -1225,6 +1226,53 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     return this.handleSendPreparedTransaction(tx);
   }
 
+  /**
+   * Call get token details API
+   *
+   * @param tokenId Token uid to get the token details
+   *
+   * @memberof HathorWalletServiceWallet
+   * @inner
+   */
+  async getTokenDetails(tokenId: string) {
+    const response = await walletApi.getTokenDetails(this, tokenId);
+    const details: TokenDetailsObject = response.details;
+
+    // We should format the response to the same format the thin wallet currently responds
+    const {
+      totalSupply,
+      totalTransactions,
+      availableAuthorities,
+      tokenInfo,
+    } = details;
+
+    const [mint, melt] = availableAuthorities.reduce((acc: [Utxo[], Utxo[]], authority) => {
+      const [mintList, meltList] = acc;
+      if (authority.authorities === TOKEN_MELT_MASK) {
+        return [
+          mintList,
+          [authority, ...meltList],
+        ];
+      } else if (authority.authorities === TOKEN_MINT_MASK) {
+        return [
+          [authority, ...mintList],
+          meltList,
+        ];
+      }
+
+      return acc;
```

Comment:
We should never reach here, right?

Reply:
We should never reach, but if we do for some weird reason, we will ignore this utxo

I'll add a comment

Reply:
Maybe we should throw an error so we can get possible problems in the wallet service response

Reply:
We don't need this anymore as the API has been refactored to return mint and melt as booleans

Refactored in aaa03b768fd1c2dfa544fe73f32e58e2512bc3fe